ztp/policygenerator/policyGen/policyBuilder.go

```diff
@@ -320,9 +331,12 @@ func (pbuilder *PolicyBuilder) createAcmPolicy(name string, namespace string, re
 	for idx, resource := range resources {
 		objTemp := BuildObjectTemplate(resource)
 		objTempArr[idx] = objTemp
+
+		if err = SetPolicyDeployWave(acmPolicy.Metadata, len(objTempArr), resource); err != nil {
```

Comment:
Maybe double-check the use of len(objTempArr) here. On line 329 the array is constructed using the len(resources). I believe this will call SetPolicyDeployWave with the same value for this second arg on each iteration. From the code below it looks like you want to call it with '1' the first iteration and any other value on subsequent iterations. Maybe pass "idx+1"?

Reply:
You are obsoultely right! Thanks for finding this out. Updated.